_posts/2022-10-10-changelog-180.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #180"
+date: 2022-10-10 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#9412] Support 2022.3 EAPs
+
+* Completion improvements:
+
+  * [#9308] Complete [custom derive] procedural macros and import the macros which aren't already in the scope
+
+    TODO: gif
+
+  * [#9332] Suggest out-of-scope modules in completion list. Previously, completion list contained only modules from the current scope
+
+    TODO: gif
+
+  * [#9313] Completion of [function-like](https://doc.rust-lang.org/reference/procedural-macros.html#function-like-procedural-macros) procedural macros now properly adds `!`
+    and doesn't show redundant info about token stream argument
+
+* [#7597] Automatically import unresolved items after pasting Rust code into the editor.
+  You can disable this behaviour via `Insert imports on paste` option in `Preferences | Editor | General | Auto Import | Rust` settings (by [@Kobzol])
+
+  TODO: gif
+
+* [#5866] Add `Wrong associated type arguments` inspection to check associated type bindings,
+  in particular [E0191](https://doc.rust-lang.org/error-index.html#E0191) and [E0220](https://doc.rust-lang.org/error-index.html#E0220) compiler errors (by [@Kobzol])
+
+  TODO: gif
+
+* [#9153] Add `Add import for path in pattern` intention to import constants in match patterns
+
+* [#9335] Provide `Create associated function` intention for path `Self::func()`
+
+* [#9376] Drop support for 2022.1 platform. Upgrade your IDE to 2022.2 to get new updates of the plugin
+
+## Fixes
+
+* [#9461] Fix sporadic false positive errors and absense of completion related to plugin's inability to find proper `impl` block.
+  The bug was introduced by [#9229] in previous release
+
+* [#9434] Don't prioritize functions that return `!` type (e.g. `panic()`) in completion list
+
+* [#9428] Fix expansion of [format](https://doc.rust-lang.org/std/fmt) macros like `println!` when parameter name is `self` keyword or syntax is incorrect
+
+* [#9293], [#9344] Fix adding imports by [Change signature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#change-sign) refactoring
+
+* [#9336] Don't suggest to import items in `derive` attribute if they are not [custom derive] macros
```

Comment:
```suggestion
* [#9336] Suggest importing only [custom derive] macros inside the `derive` attribute
```

Reply:
👍

Reply:
Although, it's not 100% true since `Debug` is not custom derive macros but we suggest it as well